hexl/thread-pool/thread-pool.hpp

```diff
@@ -0,0 +1,309 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <vector>
+
+#include "hexl/logging/logging.hpp"
+#include "hexl/util/check.hpp"
+#include "thread-pool/thread-pool-util.hpp"
+
+namespace intel {
+namespace hexl {
+
+using std::chrono::duration_cast;
+typedef std::function<void(size_t id, size_t threads)> tp_task_t;
+
+class ThreadPool {
+ private:
+  // Properties
+  size_t total_threads = 0;                     // Total number of threads
+  std::atomic_uint64_t next_thread{0};          // Points to next free thread
+  std::vector<thread_info_t*> thread_handlers;  // Thread's info
+  std::mutex pool_mutex;                        // Control pool's edition
+  bool setup_done = false;                      // Is thread pool initialized
+  inline static thread_local bool child;        // True if on child thread
```

Comment:
```suggestion
  static thread_local bool child;        // True if on child thread
```

Reply:
I get an undefined reference error if I don't use inline.

Reply:
woops not a method cool